source/MainPanel.cpp

```diff
@@ -359,11 +386,28 @@ void MainPanel::ShowScanDialog(const ShipEvent &event)
 		out << "Your scanners cannot make any sense of this " + target->Noun() + "'s interior.";
 	else if(event.Type() & ShipEvent::SCAN_OUTFITS)
 	{
-		out << "This " + target->Noun() + " is equipped with:\n";
+		if(!target->Outfits().empty())
+			out << "This " + target->Noun() + " is equipped with:\n";
+		else
+			out << "This " + target->Noun() + " is not carrying any outfits.\n";
+
+		// Split target->Outfits() into categories, then iterate over them in order.
+		map<string, map<const Outfit *, int>, CategoryNameCompare> outfitsByCategory;
 		for(const auto &it : target->Outfits())
-			if(it.first && it.second)
-				out << "\t" << it.second << " "
-					<< (it.second == 1 ? it.first->Name() : it.first->PluralName()) << "\n";
+			outfitsByCategory[it.first->Category()][it.first] = it.second;
+		for(const auto &it : outfitsByCategory)
+		{
+			if(it.second.empty())
+				continue;
+
+			// Print the category's name and outfits in it.
+			out << "\t" << it.first << "\n";
+			for(const auto &it2 : it.second)
```

Comment:
Question: what would be a good way to sort this iteration by outfit name? I see a few different ways to implement this but I'm not sure which to choose.

Reply:
I don't have much time to really look at it, but the exact same sorting is happening inside of mapOutfitterPanel so you should probably look how its done there

Reply:
That class uses a map category names to vectors that are manually sorted:

https://github.com/endless-sky/endless-sky/blob/f40324f34cbf68bac523a55d6d33e1034545c630/source/MapOutfitterPanel.h#L57-L59
https://github.com/endless-sky/endless-sky/blob/f40324f34cbf68bac523a55d6d33e1034545c630/source/MapOutfitterPanel.cpp#L289-L292

Reply:
Probably what makes sense is giving your nested `map<const Outfit *, int>` its own comparator. Then it's sorted as you construct it, instead of needing to sort it later.

Since we aren't dealing with a potentially large number of different outfits (unlike the outfitter panel), sorting on construction is fine.

Reply:
Such a comparator would essentially be a duplicate of the lambda function used in the quoted portion of MapOutfitterPanel.cpp, though?

Reply:
Is that problematic? If you find they are exact duplicates in implementation, you could write a name comparator class and use it for both.

Reply:
Added in the last push